app/services/lms_sync_service.py

```diff
@@ -0,0 +1,241 @@
+from datetime import datetime, timedelta
+import json
+from fastapi import Depends, HTTPException
```

Comment:
Don't raise HTTPException. Create some generic 500 canvas service error or something instead. You can instantiate exceptions that inherit CustomException with your own message as the first argument.

Reply:
done